src/test/java/com/example/onlinebookstore/service/BookServiceImplTest.java

```diff
@@ -0,0 +1,194 @@
+package com.example.onlinebookstore.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.example.onlinebookstore.dto.BookDtoWithoutCategoryIds;
+import com.example.onlinebookstore.dto.BookRequestDto;
+import com.example.onlinebookstore.dto.BookResponseDto;
+import com.example.onlinebookstore.exception.EntityNotFoundException;
+import com.example.onlinebookstore.mapper.BookMapper;
+import com.example.onlinebookstore.model.Book;
+import com.example.onlinebookstore.model.Category;
+import com.example.onlinebookstore.repository.BookRepository;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+
+@ExtendWith(MockitoExtension.class)
+class BookServiceImplTest {
```

Comment:
let's also add several negative cases, when book does not exist, and assert an exception with it's message, same for category